pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/PrometheusMetricsGenerator.java

```diff
@@ -131,14 +131,19 @@ private static void generateBrokerBasicMetrics(PulsarService pulsar, SimpleTextO
         parseMetricsToPrometheusMetrics(new ManagedLedgerCacheMetrics(pulsar).generate(),
                 clusterName, Collector.Type.GAUGE, stream);
 
-        // generate managedLedger metrics
-        parseMetricsToPrometheusMetrics(new ManagedLedgerMetrics(pulsar).generate(),
+        // if Topic-level metrics is disabled, we shouldn't expose managed ledger metrics
+        if (pulsar.getConfiguration().isExposeTopicLevelMetricsInPrometheus()) {
```

Comment:
I think this would break the current metrics because the namespace aggregation happens in that class for multiple metrics. 

The bigger problem with these metrics seems to be that the metric names with the bucket suffix are generated each time, and worse, using `String.format()` which is highly inefficient.

https://github.com/apache/pulsar/blob/6704f12104219611164aa2bb5bbdfc929613f1bf/pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/metrics/AbstractMetrics.java#L79-L87

Instead, we should be pre-generating these metrics/buckets strings like in https://github.com/apache/pulsar/blob/6704f12104219611164aa2bb5bbdfc929613f1bf/pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/NamespaceStats.java#L46-L63

Reply:
@merlimat The generated Strings are already cached. I fixed that in #9735 .

Reply:
Yes, I reviewed that and forgot about it. I only found out after making the same change and getting conflicts with master. :(